src/Uno.UWP/Devices/Geolocation/Geolocator.Android.cs

```diff
@@ -1,186 +1,346 @@
+
+			var startDate = DateTimeOffset.UtcNow;
+			foreach (string locationProvider in providers)
+			{
+				var location = _locationManager.GetLastKnownLocation(locationProvider);
+				if (location != null)
+				{
+					// check how old is this fix
+					var date = DateTimeOffset.FromUnixTimeMilliseconds(location.Time);
+					if (date + maximumAge > startDate)
+					{   // can be used, but is it best accuracy?
+						if (location.HasAccuracy)
+						{
+							if (location.Accuracy < bestAccuracy)
+							{
+								bestAccuracy = (int)location.Accuracy;
+								bestLocation = location;
+							}
+						}
+						else
+							bestLocation = location;
+					}
+				}
+			}
+
+			if (bestLocation != null)
+			{
+				RemoveUpdates();
+				BroadcastStatus(PositionStatus.Ready);
+				return bestLocation.ToGeoPosition();
+			}
+
+			// wait for fix
+			for (int i = (int)(timeout.TotalMilliseconds / 250.0); i > 0; i--)
+			{
+				await Task.Delay(250);
+				if (_locChanged)
+				{
+					RemoveUpdates();
+					BroadcastStatus(PositionStatus.Ready);
+					return _location.ToGeoPosition();
+				}
+			}
+
+			BroadcastStatus(PositionStatus.Disabled);
+			RemoveUpdates();
+			throw new TimeoutException("Timeout in GetGeopositionAsync(TimeSpan,TimeSpan)");
+
+		}
+
+		~Geolocator()
+        {
+			RemoveUpdates();
+			_locationManager?.Dispose();
+        }
+
+		private void RemoveUpdates()
+		{
+			if (this is null)
+			{ // when caled from destructor
+				return;
+			}
+			_locationManager?.RemoveUpdates(this);
+		}
+
+		private void RequestUpdates()
+		{
+			if (_desiredAccuracyInMeters.HasValue)
+			{
+				if (_desiredAccuracyInMeters.Value < 100)
+				{
+					_locationCriteria.HorizontalAccuracy = Accuracy.High;
+				}
+				else
+					if (_desiredAccuracyInMeters.Value < 500)
+				{
+					_locationCriteria.HorizontalAccuracy = Accuracy.Medium;
+				}
+				else
+				{
+					_locationCriteria.HorizontalAccuracy = Accuracy.Low;
+				}
+
+			}
+			else
+			{
+				_locationCriteria.HorizontalAccuracy = Accuracy.Medium;
+			}
+
+			_locationManager?.RequestLocationUpdates(_reportInterval, (float)_movementThreshold, _locationCriteria, this, Looper.MainLooper);
+		}
+
+		private uint _reportInterval = 1000;
+		private double _movementThreshold = 0;
+
+		static bool _locChanged = false;
+		static Location _location;
+
+		private Criteria _locationCriteria = new Android.Locations.Criteria() { HorizontalAccuracy = Accuracy.Medium };
+
+
+		public uint ReportInterval
+		{
+			get => _reportInterval;
+			set
+			{
+				_reportInterval = value;
+				RemoveUpdates();
+				RequestUpdates();
+			}
+		}
+
+		public double MovementThreshold
+		{
+			get => _movementThreshold;
+			set
+			{
+				_movementThreshold = value;
+				RemoveUpdates();
+				RequestUpdates();
+			}
+		}
+
+		public uint? DesiredAccuracyInMeters
+		{
+			get => _desiredAccuracyInMeters;
+			set
+			{
+				_desiredAccuracyInMeters = value;
+				RemoveUpdates();
+				RequestUpdates();
+			}
+		}

```

Comment:
Don't replace the common code. You need to add a partial method in the common code so that you can invoke `RemoveUpdates();` and `RequestUpdates();` in the android specific code.

Reply:
partial set in common code?

Reply:
Can you be more specific about this partial method?

Reply:
Can you be more specific about this "partial method in the common code"?

Reply:
Don't remove the method in the common code, meaning the code compiled for all targets.

Reply:
Difference between common and Android code is to reset updates. I don't know how to leave setter in common code, and for Android make these two calls within it.
Do you want to make something like `partial void ResetUpdates();` in common code, and for Android define it as
`
  partial void ResetUpdates();
  {
	RemoveUpdates();
	RequestUpdates();
  }
`

?

Reply:
Don't copy the `DesiredAccuracyInMeters` property, and alter the one in the common code to add a partial method name `OnDesiredAccuracyInMetersChanged` and do your RemoveUpdates/RequestUpdates there.